ci/cloudbuild/builds/universe-domain.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+#
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+source "$(dirname "$0")/../../lib/init.sh"
+source module ci/lib/io.sh
+source module ci/cloudbuild/builds/lib/bazel.sh
+source module ci/cloudbuild/builds/lib/cloudcxxrc.sh
+source module ci/cloudbuild/builds/lib/universe_domain.sh
+
+export CC=clang
+export CXX=clang++
+
+if [[ -n "${UD_SA_KEY_FILE}" ]]; then
```

Comment:
When doesn't this exist? Looks like when there is no UD_SERVICE_ACCOUNT secret.

Should the build just fail if it doesn't exist?

Reply:
If we run the `domain-universe-pr` trigger locally, it will run into the branch https://github.com/googleapis/google-cloud-cpp/blob/acf5b1cb8c20c4b552bc898214af80ea83b48822/ci/cloudbuild/builds/universe-domain-demo.sh#L33-L37

In this case, it cannot read the secret, so it actually uses the non-universe-domain default secrets.

Reply:
Thanks